### ISSUE TITLE
Notice: Undefined index: contact_link 

### ISSUE THREAD

**Issue by sudama108** [mecachisenros/caldera-forms-civicrm#95]
Hi There,

I get these errors in my form when it loads in our website. We are running:
CiviCRM 5.3.1
Caldera Forms 1.7.2
Caldera Forms CiviCRM 0.4.4

Notice: Undefined index: contact_link in .../wp-content/plugins/caldera-forms-civicrm-master/processors/contact/class-contact-processor.php on line 739

Notice: Undefined index: contact_link in .../wp-content/plugins/caldera-forms-civicrm-master/processors/contact/class-contact-processor.php on line 782

Notice: Undefined index: contact_link in .../wp-content/plugins/caldera-forms-civicrm-master/processors/contact/class-contact-processor.php on line 825

These errors do not show when I log into CiviCRM and then browse to our form on the website.
Kindly help with resolving.

Thanks,
regards
Sudesh

**Comment by danaskallman** [comment]
@sudama108 Caldera Forms should be 1.7.3 or above, can you update the plugin and test?

**Comment by mecachisenros** [comment]
@sudama108 I'll get it sorted, although it doesn't affect the plugin functionality, it is annoying and makes noise in your logs.

Question, are you seeing the notices in the front end (public facing pages)?

**Comment by sudama108** [comment]
@mecachisenros Yes, i'm seeing these in the front end (public facing pages) of our site. Thanks for the support.

**Comment by mecachisenros** [comment]
Do you happen to have WP_DEBUG enabled (`define( 'WP_DEBUG', true )`) in your wp-config.php file? If so I would encourage you to disable it in production, if for any reason you do need debugging in production, disable the display of notices and warnings with `define( 'WP_DEBUG_DISPLAY', false )` and enable logging instead `define( 'WP_DEBUG_LOG', true )`.

More info on debugging in WordPress [here](https://codex.wordpress.org/Debugging_in_WordPress).

**Comment by sudama108** [comment]
I currently have the following settings in my wp-config.php file:

ini_set('display_errors','Off');
ini_set('error_reporting', E_ALL );
define('WP_DEBUG', false);
define('WP_DEBUG_DISPLAY', false);

Therefore, I find it strange to still so those notices appear.

**Comment by kcristiano** [comment]
@sudama108 There are some hosts that I have run into that have `display_errors` set to on.   I would check this and set `display_errors = Off` if that is not the value.

**Comment by mecachisenros** [comment]
In theory the config order is like this:
1. `php.ini`
2. `.htaccess` / Apache directive
3. `wp-config.php`

I would double check php.ini and .htaccess files, or if your hosting uses some control panel (cPanel and alike) check its settings, and last I would do a `grep -r 'display_errors' /path/to/wordpress/directory/wp-content` and see if any plugin is overriding your settings.

Also I would set `error_reporting` to something like `E_ALL & ~E_NOTICE & ~E_DEPRECATED & ~E_STRICT` in production.

@kcristiano funny enough `display_errors` comes enabled by default, from php.ini
```
; Possible Values:
;   Off = Do not display any errors
;   stderr = Display errors to STDERR (affects only CGI/CLI binaries!)
;   On or stdout = Display errors to STDOUT
; Default Value: On
; Development Value: On
; Production Value: Off
; http://php.net/display-errors
```

**Comment by sudama108** [comment]
An update on this issue:  I contacted my web hosting support. This was their response: 
It seems that the custom error handler of CiviCRM, or its plugin, overrides the php.ini settings, but we could not find a way to fully disable error reporting for all scripts/plugins. Unfortunately, php.ini settings cannot override custom PHP code. To resolve the issue at hand, we added the following in the beginning of .../wp-content/plugins/caldera-forms-civicrm-master/processors/contact/class-contact-processor.php, right after the PHP opening tag:
<?php
error_reporting(0);

The form no longer shows the  PHP errors/notices.
